core/src/wsv.rs

```diff
@@ -367,6 +412,37 @@ impl<W: WorldTrait> WorldStateView<W> {
         &self.world.domains
     }
 
+    /// Get `Domain` and pass it to closure.
+    ///
+    /// # Errors
+    /// Fails if there is no domain
+    pub fn map_domain<T>(
+        &self,
+        id: &<Domain as Identifiable>::Id,
+        f: impl FnOnce(&Domain) -> T,
```

Comment:
Not saying that we should, but I'd consider `Infallible` in the future. 

https://doc.rust-lang.org/std/convert/enum.Infallible.html